lib/HunkView.ml

```diff
@@ -37,25 +40,43 @@ let split_and_align_hunk hunks : line list * line list =
   in
   process_hunk [] [] hunks
 
-(* Normal Mode *)
+(* Styling Functions *)
+
+let style_text (text : string) (attr : Notty.attr) (mode : rendering_mode) :
+    Ui.t =
+  match mode with Color -> W.string ~attr text | TextMarkers -> W.string text
+
+let style_word (word : string)
+    (change_type : [ `Added | `Removed | `Unchanged ]) (mode : rendering_mode) :
+    Ui.t =
+  let styled_word =
+    match (mode, change_type) with
+    | Color, `Added -> W.string ~attr:Notty.A.(fg green) word
+    | Color, `Removed -> W.string ~attr:Notty.A.(fg red) word
+    | Color, `Unchanged -> W.string word
+    | TextMarkers, `Added -> added_marker (W.string word)
+    | TextMarkers, `Removed -> removed_marker (W.string word)
+    | TextMarkers, `Unchanged -> W.string word
+  in
+  Ui.hcat [ styled_word; W.string " " ]
```

Comment:
This is adding an extra space at the end of the line. However, this small bug is not introduced in this PR and should be fixed in another one.

Reply:
Ok